src/solver/conv_hip_implicit_gemm_bwd_data_v4r1_xdlops.cpp

```diff
@@ -154,38 +184,56 @@ PerformanceImplicitGemmBwdDataV4R1Xdlops::CalculateGemmBBlockCopyPerformancePara
         }
 
         // calculate threadwise copy size
-        int b_data_per_thread_copy = (GemmKPerBlock * GemmNPerBlock) / BlockSize;
+        int b_data_per_thread_copy = (GemmKPerBlock * GemmNPerBlock * GemmKPACKSize) / BlockSize;
 
         if(!(b_data_per_thread_copy > 0))
             MIOPEN_THROW("invalid performance parameter");
 
         // GemmBBlockCopySrcDataPerRead_GemmN also bounded by size of threadwise copy
-        SrcDataPerRead_GemmN = gcd(SrcDataPerRead_GemmN, b_data_per_thread_copy);
+        SrcDataPerRead_GemmN                  = gcd(SrcDataPerRead_GemmN, b_data_per_thread_copy);
```

Comment:
GemmN = merge (N, HTildaSlice, WTildaSlice). But here  the calculation of ```SrcDataPerRead_GemmN``` doesn't consider what is the value of  ```WTildaSlice```. If ```WTildaSlice``` is odd number, then we cannot do vector read on GemmN.

I'm surprised this logic has not caused any failure. We need to discuss on this logic, it seem wrong to me

Reply:
Yes ,we can dicusss it tomorrow